### PR TITLE
Added not fully integrated localization logic, improved Page Navigation.

### PR DESCRIPTION
**Pull request by J0nathan550**
Fixed: Page Navigation that before was hard coded using string. Problem start to occure when I started developing localization for the Pane strings. I just noticed that when you clicked on buttons (in Pane) and they have different label in them it would not switch page because of the hardcoded Pane string content. It was fixed by adding events.

Added localization logic, the code itself works just fine with no problems. What is left is to bind everywhere the strings with localization we desire. I will continue working on the localization binding. But this is my safe push to keep up with other people. It should cost no problem in code whatsover.

Works by using DynamicResource binding, it grabs from Avalonia Resources specific file of language that you use (default is English) and translates it.

Pages translated: PlayView, MainView (Pane).

What to finish: Fully bind everywhere (where possible strings to translate), create big template English resource file to later add translation for other languages. Implement same localization binding but for the .cs source code (strings inside of source code are not translated, but I hope I will figure out soon how to translate them.).

Video demonstration of this PR:
https://youtu.be/XgD_EiMDU9Y

### CODE REVIEW

File: SIT.Manager.Avalonia/Models/LocalizationConfig.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/Models/LocalizationConfig.cs:53]: Models should contain minimal to no logic inside of them. This should be pushed into a service

File: SIT.Manager.Avalonia/Models/LocalizationConfig.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/Models/LocalizationConfig.cs:7]: I'm personally not a fan of marking the localization system as static and accessing it globally. This would be better added as a service and injected into classes that need it

File: SIT.Manager.Avalonia/Models/LocalizationConfig.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/Models/LocalizationConfig.cs:9]: We should probably use CultureInfo over our own enums

File: SIT.Manager.Avalonia/ViewModels/MainViewModel.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/ViewModels/MainViewModel.cs:54]: Accessing the localization system like this gives me the ick, see previous comments about services

File: SIT.Manager.Avalonia/Views/MainView.axaml.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/Views/MainView.axaml.cs:83]: I don't see how this is any better than the current system. In the end it's still hard-coded types and events you're just moving where the hard coded types lie

File: SIT.Manager.Avalonia/Views/SettingsPage.axaml
Comment (Plootie) [inline SIT.Manager.Avalonia/Views/SettingsPage.axaml:37]: These item's should be loaded in from a list using data templating for maintainability

File: SIT.Manager.Avalonia/Views/SettingsPage.axaml
Comment (Plootie) [inline SIT.Manager.Avalonia/Views/SettingsPage.axaml:34]: I think we should stay away from using an event here. Use IPropertyChanged to notify when the SelectedItem is changed

File: SIT.Manager.Avalonia/Views/SettingsPage.axaml.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/Views/SettingsPage.axaml.cs:18]: See previous comment about avoid events

**Review by Plootie — CHANGES_REQUESTED** [review]

File: SIT.Manager.Avalonia/Views/MainView.axaml.cs
Comment (J0nathan550) [reply SIT.Manager.Avalonia/Views/MainView.axaml.cs:83]: > I don't see how this is any better than the current system. In the end it's still hard-coded types and events you're just moving where the hard coded types lie

Better in they way that you can switch pages that are not depended literally on string. Current system checks what current page name is. And if the page name != whatever in Dictionary we cannot go to another page. Also why to bother? They are meant to be static. I don't think we will add 10.000+ buttons in that pane what so ever.

File: SIT.Manager.Avalonia/Models/LocalizationConfig.cs
Comment (J0nathan550) [reply SIT.Manager.Avalonia/Models/LocalizationConfig.cs:9]: > We should probably use CultureInfo over our own enums

Problem behind Culture Info it's not 100% correct when you are starting the application. But still a good point to set at the very first start of the application the language based on what language you have on computer.

File: SIT.Manager.Avalonia/Views/SettingsPage.axaml.cs
Comment (J0nathan550) [reply SIT.Manager.Avalonia/Views/SettingsPage.axaml.cs:18]: I just don't know how to use commands, this thing is complicated for me.

File: SIT.Manager.Avalonia/Views/MainView.axaml.cs
Comment (Plootie) [reply SIT.Manager.Avalonia/Views/MainView.axaml.cs:83]: This is actually my fault as I appear to have accidentally swapped from using the NavigationViewItem's Tag property to the InvokedItem which gives the display name. The Tag property allows us to have an identifier separate from the display value without needing an event for each item. I will push the needed changes for this now

**Comment by Plootie** [comment]
I've made a [pull request](https://github.com/stayintarkov/SIT.Manager.Avalonia/pull/34) to fix the NavigationView problem. Once Artehe signs off on it I'll merge it into master for you to update from

**Comment by J0nathan550** [comment]
> I've made a [pull request](https://github.com/stayintarkov/SIT.Manager.Avalonia/pull/34) to fix the NavigationView problem. Once Artehe signs off on it I'll merge it into master for you to update from

I saw that thank you, I will now avoid any events with this one 😄

**Comment by J0nathan550** [comment]
I'm still working, I just fixed issue with my branch. No review required.

**Comment by J0nathan550** [comment]
- Implemented MVVM, Service of Localization.
- Translated almost fully pages.
- Added just logic to be able Translate source code, still I have to translate entire source code. 
Hope you will like it.

**Review by Plootie — CHANGES_REQUESTED** [review]
I'll hold off reviewing the code for now. This appears to be broken on my end. Selecting any language overwrites the option with "English (United States)" 
bNwEhVV4f7

**Comment by J0nathan550** [comment]
I did one dumb mistake, sorry one second.

**Comment by Plootie** [comment]
I'll review this in the morning, There's quite a few things I wanna comment on that'll take too long tonight

File: SIT.Manager.Avalonia/Views/MainWindow.axaml
Comment (Plootie) [inline SIT.Manager.Avalonia/Views/MainWindow.axaml:16]: Not sure why this is here. It's possible that these options aren't needed but I'm not a fan of changing things unrelated to the PR and undocumented

File: SIT.Manager.Avalonia/Views/SettingsPage.axaml
Comment (Plootie) [inline SIT.Manager.Avalonia/Views/SettingsPage.axaml:39]: This can be condense into `<Grid ColumnDefinitions="auto, *">`

File: SIT.Manager.Avalonia/Views/PlayPage.axaml
Comment (Plootie) [inline SIT.Manager.Avalonia/Views/PlayPage.axaml:24]: Removal of the width attributes here causes the textboxs to shrink to minimum size when you enter a single character as its resizing to largest content and once a character is entered the watermark disappears causing the shrink

File: SIT.Manager.Avalonia/Models/LocalizationModel.cs
Comment (J0nathan550) [reply SIT.Manager.Avalonia/Models/LocalizationModel.cs:1]: But you cannot store everything in CultureInfo, because it's gonna be a waste of memory, better to store few variables. Also it's gonna be for me at least a lot harder to understand and save literally CultureInfo.

File: SIT.Manager.Avalonia/Views/MainWindow.axaml
Comment (J0nathan550) [reply SIT.Manager.Avalonia/Views/MainWindow.axaml:16]: I think it's better to remove AcrylicBlur because we literally do not use transparent AcrylicBlur background, it can make a difference in perfomance when you render a window.

File: SIT.Manager.Avalonia/Views/SettingsPage.axaml
Comment (J0nathan550) [reply SIT.Manager.Avalonia/Views/SettingsPage.axaml:39]: Sugar syntax but okay 😄

File: SIT.Manager.Avalonia/Models/LocalizationModel.cs
Comment (J0nathan550) [reply SIT.Manager.Avalonia/Models/LocalizationModel.cs:1]: But I will see, maybe it's not that hard.

File: SIT.Manager.Avalonia/Views/MainWindow.axaml
Comment (Plootie) [reply SIT.Manager.Avalonia/Views/MainWindow.axaml:16]: Can we move this to another PR. I don't disagree that these are possibly useless but it's outside the scope of this PR and I don't like making undocumented changes

File: SIT.Manager.Avalonia/Models/ManagerConfig.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/Models/ManagerConfig.cs:31]: This falls back to the users current culture info, but if the user's culture doesn't match any of the localizations we have the selected field will be blank (In my case this defaults to en-GB, since that doesn't match any localizations we have the selected item becomes blank)

File: SIT.Manager.Avalonia/Services/LocalizationService.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/Services/LocalizationService.cs:74]: These two methods have significant overlap, Push some of the duplicate code to another function to avoid repeating the same thing. Also instead of using a try catch were reference the known available localizations to ensure it's valid before attempting

File: SIT.Manager.Avalonia/Services/LocalizationService.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/Services/LocalizationService.cs:117]: I don't like this approach. We're repeating ourselves massively here. We already pull values such as `SIT.Manager.Avalonia.Localization.en-US.axaml` from the first half of the LINQ expression, only to extract the country code, convert it into a CultureInfo to the reconstruct that exact path again later to create the resource

File: SIT.Manager.Avalonia/ViewModels/SettingsPageViewModel.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/ViewModels/SettingsPageViewModel.cs:125]: This will call the localization code on every property change regardless of if the localization was actually changed

**Review by Plootie — CHANGES_REQUESTED** [review]

File: SIT.Manager.Avalonia/Models/ManagerConfig.cs
Comment (J0nathan550) [reply SIT.Manager.Avalonia/Models/ManagerConfig.cs:31]: Yeah, I forgot about that issue will fix it.

File: SIT.Manager.Avalonia/ViewModels/SettingsPageViewModel.cs
Comment (J0nathan550) [reply SIT.Manager.Avalonia/ViewModels/SettingsPageViewModel.cs:125]: Sure, sorry, I will add propertychange based on if the Settings current localization was changed.

File: SIT.Manager.Avalonia/Services/LocalizationService.cs
Comment (Plootie) [inline SIT.Manager.Avalonia/Services/LocalizationService.cs:1]: My only gripe left with this PR is some of these methods. I'm seeing a lot of `avares://SIT.Manager.Avalonia/Localization/*.axaml` that eventually ends up getting passed down to `CreateResourceLocalization`. This could be changed so that only the file name gets passed in to the method and it constructs the rest of the path itself. Once that's addressed I'm happy to merge this ^^

**Review by Plootie — CHANGES_REQUESTED** [review]

File: SIT.Manager.Avalonia/Services/LocalizationService.cs
Comment (J0nathan550) [reply SIT.Manager.Avalonia/Services/LocalizationService.cs:1]: Do you want me to later recreate PR to avoid thousands of commits?

**Comment by J0nathan550** [comment]
Done!

**Review by Plootie — APPROVED** [review]
I think I'm happy with this :3 @artehe Do you mind just giving this a skim to make sure I haven't missed something dumb?

File: SIT.Manager.Avalonia/ViewModels/SettingsPageViewModel.cs
Comment (artehe) [inline SIT.Manager.Avalonia/ViewModels/SettingsPageViewModel.cs:129]: I know I said two things but I can't think how to handle the second so we can leave it for now and I'll create an issue to come back to it sometime in the future.  So one last thing then this is good to go. 

Could we take the highlighted lines out and instead use th the specific property change value for CurrentLanguageSelected?

If you wanna just copy and paste the function it'd be this:
```
    partial void OnCurrentLocalizationChanged(CultureInfo value)
    {
        _localizationService.Translate(value);
        Config.CurrentLanguageSelected = value.Name;
    }
```

**Review by Plootie — APPROVED** [review]

**Pull request merged**
